api/components/com_users/src/Controller/UsersController.php

```
@@ -65,4 +87,91 @@ protected function save($recordKey = null)
+		$filter        = InputFilter::getInstance();
+
+		if (array_key_exists('state', $apiFilterInfo))
+		{
+			$this->modelState->set('filter.state', $filter->clean($apiFilterInfo['state'], 'INT'));
+		}
+
+		if (array_key_exists('active', $apiFilterInfo))
+		{
+			$this->modelState->set('filter.active', $filter->clean($apiFilterInfo['active'], 'INT'));
+		}
+
+		if (array_key_exists('groupid', $apiFilterInfo))
+		{
+			$this->modelState->set('filter.group_id', $filter->clean($apiFilterInfo['groupid'], 'INT'));
+		}
+
+		if (array_key_exists('search', $apiFilterInfo))
+		{
+			$this->modelState->set('filter.search', $filter->clean($apiFilterInfo['search'], 'STRING'));
+		}
+
+		if (array_key_exists('registrationdate', $apiFilterInfo))
+		{
+			$rangeFilter = $filter->clean($apiFilterInfo['registrationdate'], 'STRING');
+
+			if (!in_array($rangeFilter, $this->supportedRange))
+			{
+				// Send the error response
+				$this->sendResponse('registrationdate', 400);
+			}
+
+			$this->modelState->set('filter.range', $rangeFilter);
+		}
+
+		if (array_key_exists('lastvisitdate', $apiFilterInfo))
+		{
+			$rangeFilter = $filter->clean($apiFilterInfo['lastvisitdate'], 'STRING');
+
+			if (!in_array($rangeFilter, $this->supportedRange))
+			{
+				// Send the error response
+				$this->sendResponse('lastvisitdate', 400);
+			}
+
+			$this->modelState->set('filter.lastvisitrange', $rangeFilter);
+		}
+
+		return parent::displayList();
+	}
+
+	/**
+	 * Send the given data as JSON response in the following format:
+	 *
+	 * {"success":true,"message":"ok","messages":null,"data":[{"type":"dir","name":"banners","path":"//"}]}
+	 *
+	 * @param   string   $field         The wrong field
+	 * @param   integer  $responseCode  The response code
+	 *
+	 * @return  void
+	 *
+	 * @since   4.0.0
+	 */
+	private function sendResponse(string $field = null, int $responseCode = 200): void
+	{
+		// Set the correct content type
+		$this->app->setHeader('Content-Type', 'application/json');
```

This will not be a valid JSON API response. You need to throw an InvalidParameterException from the library https://github.com/tobyzerner/json-api-php/blob/master/src/Exception/InvalidParameterException.php